command/agent/consul/connect.go

```diff
@@ -120,7 +117,23 @@ func connectSidecarRegistration(serviceId string, css *structs.ConsulSidecarServ
 				Name:         "Connect Sidecar Aliasing " + serviceId,
 				AliasService: serviceId,
 			},
-		},
+		}
+	} else {
+		// insert a NOOP check to avoid Consul inserting the default
```

Comment:
This comment is a bit misleading - the aliasing here is functional to consul service discovery - e.g. if the tcp check exists and is failing, queries for the aliased service will not return results including this sidecar (i.e. when using connect upstreams).